### PR TITLE
feat: add goreleaser workflow

### PR DESCRIPTION
**Pull request by eddycharly**
This PR adds goreleaser workflow and config.

### CODE REVIEW

**Comment by eddycharly** [comment]
cc @apelisse

File: .goreleaser.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    - go mod tidy
+
+builds:
+  - env:
+      - CGO_ENABLED=0
+    goos:
+      - linux
+      - windows
+      - darwin
+    binary: kyverno-json
```
Comment (alexzielenski) [inline .goreleaser.yaml:12]: probably should change

**Comment by eddycharly** [comment]
(i didn't try it, could have typos in the config 🤞)

**Review by alexzielenski — CHANGES_REQUESTED** [review]
Thanks so much for this! Definitely what we were just talking about needing...

File: .goreleaser.yaml (hunk 1 above)
Comment (eddycharly) [reply .goreleaser.yaml:12]: yeah, just saw it 🤦

Comment (eddycharly) [reply .goreleaser.yaml]: fixed

**Comment by eddycharly** [comment]
@alexzielenski let me know if you want more config, goreleaser can use cosign, build/sign images with ko, publish on brew, etc...

**Comment by caesarxuchao** [comment]
Hi all, thanks for starting this PR. What's required to get this PR merged? It would be very helpful to have released binaries of kubectl-validator.

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      contents: write
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+        with:
+          fetch-depth: 0
+      - name: Fetch all tags
+        run: |
+          set -e
```
Comment (apelisse) [inline .github/workflows/release.yaml:20]: What's the point of that `set -e`, will the step not fail if `git fetch` fails anyway?

**Review by apelisse — COMMENTED** [review]
I'm fine to merge this, test and see how it goes, we don't have a lot to lose if it doesn't work perfectly.

**Comment by apelisse** [comment]
OK, let's try it now!

**Comment by apelisse** [comment]
I just pushed v0.0.2 (from the same commit as v0.0.1) and it doesn't look like it's triggering: https://github.com/kubernetes-sigs/kubectl-validate/actions/workflows/release.yaml

**Comment by caesarxuchao** [comment]
Thanks for trying! Our company is using a forked release so we are not blocked. We'll shift to the official release whenever that's ready.
kubectl-validate has caught numerous issues in our dev workflow. Thanks for creating this super useful tool!

**Pull request merged**